Add courses model

MahmoudSerag opened the pull request:
- Create course schema
- Add course validation messages
- Add relation between course  and mentor
- Add relation between course and review

Commit:
Add courses model
- Create course schema
- Add course validation messages
- Add relation between course  and mentor
- Add relation between course and review

IslamGoher commented on models/courses.js:
```diff
@@ -0,0 +1,68 @@
+    type: String,
+    required: [true, 'Please add a description']
+  },
+  picture: {
+    type: String,
+    required: [true, 'Please add a picture']
+  },
```
please add url match

IslamGoher commented on models/courses.js:
```diff
@@ -0,0 +1,68 @@
+    default: 0
+  },
+  topicsList: {
+    type: String,
```
please convert lopicslist to an array

IslamGoher commented on models/courses.js:
```diff
@@ -0,0 +1,68 @@
+      required: [true, 'Please add a review id']
+    }
+  ],
+  mentorId: [
```
please change mentorId data type to ObjectId not array of ObjectIds

IslamGoher reviewed (changes requested):
please fix those problems

Commit:
Edit course model
- Add url match to picture property
- Convert topicsList property to an array
- Change mentorId property to ObjectId

fix #6
